packages/repo-tools/src/commands/openapi/generate.ts

```diff
@@ -53,8 +53,9 @@ async function generate(
 // ******************************************************************
 // * THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. *
 // ******************************************************************
-
-export default ${JSON.stringify(yaml, null, 2)} as const;`,
+import {createOpenApiRouter} from '@backstage/backend-openapi-utils';
+export const spec = ${JSON.stringify(yaml, null, 2)} as const;
+export default async ()=>createOpenApiRouter<typeof spec>(spec);`,
```

Comment:
Ah, you might need the new `const` modifier for the type parameter. Tbh it probably makes sense to have it at least in the generated code either way though, to make it compatible with older TS versions

Reply:
Yeah, really cool, didn't realize that was part of 5.0. I think for now since we're still supporting TS 4, we should keep as is, but that would definitely simplify this code.